agents/unix/conf/base/conf_cpu.c

```diff
@@ -895,6 +899,81 @@ cpu_thread_grab(const char *name)
     return rc;
 }
 
+static te_errno
+memory_get(unsigned int gid, const char *oid, char *value)
+{
+  UNUSED(gid);
+  UNUSED(oid);
+
+#ifdef _SC_PHYS_PAGES
+  long n_pages, page_size;
+  uint64_t mem;
+
+  errno = 0;
+
+  n_pages = sysconf(_SC_PHYS_PAGES);
+  if (n_pages == -1)
+  {
+      ERROR("Failed to get sysconf number of memory pages");
+      return TE_OS_RC(TE_TA_UNIX, errno);
+  }
+
+  page_size = sysconf(_SC_PAGESIZE);
+  if (page_size == -1)
+  {
+      ERROR("Failed to get sysconf memory page size");
+      return TE_OS_RC(TE_TA_UNIX, errno);
+  }
+
+  /* Total memory in bytes */
+  mem = (uint64_t)n_pages * (uint64_t)page_size;
+
+  snprintf(value, RCF_MAX_VAL, "%" PRIu64, mem);
+#else
+  UNUSED(value);
+#endif
+
+  return 0;
+}
```

Comment:
I'd say it should be like this:
```
uint64_t mem = 0;
#ifdef _SC_PHYS_PAGES
....
#endif
snprintf(value, RCF_MAX_VAL, "%" PRIu64, mem);
return 0;
```

That way we're totally sure that the node returns a valid number, but if it cannot be detected - it's just `0`